internal/services/apimanagement/api_management_workspace_resource.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2024-05-01/workspace"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ApiManagementWorkspaceModel struct {
+	WorkspaceId       string `tfschema:"workspace_id"`
+	ResourceGroupName string `tfschema:"resource_group_name"`
+	ServiceName       string `tfschema:"service_name"`
+	WorkspaceName     string `tfschema:"workspace_name"`
+}
+
+type ApiManagementWorkspaceResource struct{}
+
+var _ sdk.Resource = ApiManagementWorkspaceResource{}
+
+func (r ApiManagementWorkspaceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"workspace_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9-]{1,80}$`),
+				"Workspace ID must be 1 - 80 characters long, contain only letters, numbers and hyphens.",
+			),
+			Description: "The ID of the API Management Workspace.",
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"service_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "The name of the API Management Service in which this Workspace should be created.",
+		},
+		"workspace_name": {
```

Comment:
This could be the display name
```suggestion
		"display_name": {
```